src/hooks/use-sync-mst-prop-to-firebase.ts

```diff
@@ -35,7 +37,12 @@ export function useSyncMstPropToFirebase<T extends string | number | boolean | u
   };
   const mutation = useMutation((value: T) => {
     const should = typeof shouldMutate === "function" ? shouldMutate(value) : shouldMutate;
-    return should ? firebase.ref(path).update({ [prop]: value }) : Promise.resolve();
+    const mutations = Promise.all([
+      should ? firebase.ref(path).update({ [prop]: value }) : Promise.resolve(),
+      additionalMutation ? additionalMutation(prop, value) : Promise.resolve()
+    ]);
```

Comment:
I'd guess that additionalMutation shouldn't be run unless `should` is true. So you could change this to:
```suggestion
    const mutations = should
       ? Promise.all([
           firebase.ref(path).update({ [prop]: value }),
           additionalMutation ? additionalMutation(prop, value) : Promise.resolve()
         ])
       :  Promise.resolve()
    ]);
```